File: batch_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def batch_process(list_file, tool_func, config, *args, **kwargs):
    if not check_file_permission(list_file, "read", config):
        raise PermissionError(f"Read access denied for {list_file}")
    list_file = Path(list_file).resolve()
    with open(list_file, "r", encoding="utf-8") as f:
        files = [line.strip() for line in f if line.strip()]
    results = []
    for file_path in files:
        results.append(tool_func(file_path, config, *args, **kwargs))
    return "\n".join(results)

File: ts-tools/batch_tool.ts


import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function batchProcess(listFile: string, toolFunc: (filePath: string, config: any, ...args: any[]) => Promise<string>, config: any, ...args: any[]): Promise<string> {
    return new Promise((resolve, reject) => {
        checkFilePermission(listFile, "read", config)
            .then(() => {
                const resolvedPath = path.resolve(listFile);
                fs.readFile(resolvedPath, 'utf-8', (err, data) => {
                    if (err) {
                        reject(err);
                    } else {
                        const files = data.split('\n').filter(line => line.trim() !== '');
                        const promises = files.map(file => toolFunc(file, config, ...args));
                        Promise.all(promises)
                            .then(results => resolve(results.join('\n')))
                            .catch(reject);
                    }
                });
            })
            .catch(reject);
    });
}

